thorlcr/activities/loop/thloopslave.cpp

```diff
@@ -1245,10 +1246,12 @@ class CGraphLoopResultReadSlaveActivity : public CSlaveActivity
         if ((int)sequence >= 0)
         {
             assertex(container.queryResultsGraph());
+            Owned<CGraphBase> graph;
             graph_id resultGraphId = container.queryXGMML().getPropInt("att[@name=\"_graphId\"]/@value");
-            if (!resultGraphId)
-                resultGraphId = container.queryResultsGraph()->queryGraphId();
-            Owned<CGraphBase> graph = queryJobChannel().getGraph(resultGraphId);
+            if (resultGraphId == container.queryResultsGraph()->queryGraphId())
+                graph.set(container.queryResultsGraph());
+            else
+                graph.setown(queryJobChannel().getGraph(resultGraphId));
```

Comment:
talking with @ghalliday offline a bit.. I'm not sure.. if there were grand child parallel queries that wrote results to a grand parent.. I think this might be a problem..

Looking at the roxie code, I wondered if similar there too..

May need a test example.